### PR TITLE
Lottie animation component

### PR DESCRIPTION
**Pull request by bakulf**
# First question: why?
QT has a Lottie animation component but it's hard to add it to our package (no linux packages) and, more important, it's not [implementation-completed](https://doc.qt.io/qt-5/qml-qt-labs-lottieqt-lottieanimation.html#details). @flozia did a really good job importing a 3rd-party library [lottie-qml](https://github.com/kbroulik/lottie-qml/), but there were several problems in that implementation:

- that code is not maintained. The last commit was done 3 years ago.
- the code is not production-ready: it's full of TODO comments, undocumented code, etc.
- there are no tests!
- it uses QT deprecated functions (mainly `Qt.include()`).
- it's a really really simple wrapper around [lottie-web](https://github.com/airbnb/lottie-web).
- We need a fast solution until QtLottie is ready enough. Probably it will be in Qt6.

I decided to implement our own Lottie component starting from these ideas:
- 100% test coverage (QML + unit tests)
- simple to maintain
- independent from the VPN codebase

This PR implements this new component and, because it's a lot of code, I need reviewers. Here is a description of the code:

# Lottie folder
1. The code is self-contained. It's all included in a `lottie` folder. Here you can find a `lottie.pri` file to be used in the VPN project (and internally) but also a `lottie.pro` in case you want to compile it independently. 
2. `lib`, the lottie code, as a library. This is what we care for the MozillaVPN project
3. `bin`, a tool to run lottie files
4. `tests/qml`, qml tests
5. `tests/unit`, unit tests

# Documentation
`README.md` and `lib/lottie/LottieAnimation.qml` are the 2 main files. Use them to know how to use the component.

# lottietest
in the `bin` folder, you will find a lottietest app. I use it to render the lottie files directly. You can also use it to see how to apply the library to the MozillaVPN codebase.

# Tests
1. unit-tests cover all the C++ DOM models: document, navigator, windows. I have implemented only the required methods for lottie-web.
2. qml-tests, e2e tests. All the features should be covered. If you spot something uncovered, let me know.

# MozillaVPN integration
Here there is just a little. The majority is done elsewhere. But you can find:
- LICENSE.md changes. I already contacted the legal team to know if these changes are OK
- QML/unit tests coverage. I attached the Lottie tests to the mozillaVPN ones
- Compilation: we compile the lottie module together with all the rest.

# Implementation details
- The lottie_min.js is the canvas-only file taken from the official [lottie CDN](https://cdnjs.com/libraries/bodymovin). I wrap it into a module using a `lottie_wrap.js.template`. In this simple wrapper adds a few listener when needed.
- I have a simple implementation for the DOM document, navigator, and window to make lottie-web happy.

# Alternatives
There are mainly 3 alternatives:
- QtLottie - not good in Qt5. Too many limitations
- [Skia](https://skia.org/docs/user/modules/skottie/) - a big library to be included.
- [rlottie](https://github.com/Samsung/rlottie) - this was tempting! But we could have issues with the windows compilation.

### CODE REVIEW

**Review by flozia — APPROVED** [review]
It’s amazing that we have our own Lottie implementation for QML now. I only have two things that I would like your opinion on and left inline comments for:
1.  I think it would be nice if the animation could change when the `source` property of `LottieAnimation` changes, even if it’s not currently playing.
2. In order to be able to keep track of `currentTime` and `totalTime` when the animation is paused it could be helpful to handle more states in the `update` method of `LottieStatus`.

Thank you so much for your work @bakulf!

File: lottie/lib/lottieprivate.cpp [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+void LottiePrivate::play() {
+  if (runAnimationFunction("play", QList<QJSValue>())) {
+    m_status.update(true);
+  }
+}
+
+void LottiePrivate::pause() {
+  if (runAnimationFunction("pause", QList<QJSValue>())) {
+    m_status.update(false);
```
[the remark this answers is not in the document]
Comment (bakulf) [reply lottie/lib/lottieprivate.cpp:322]: Done.

File: lottie/lib/lottieprivate.cpp [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+  bool wasPlaying = m_status.playing();
+
+  destroyAnimation();
+
+  QJSValue containerValue = engine()->toScriptValue(m_container);
+  runFunction(containerValue, "clear", QList<QJSValue>());
+
+  createAnimation();
+
+  if (wasPlaying) play();
```
[the remark this answers is not in the document]
Comment (bakulf) [reply lottie/lib/lottieprivate.cpp]: Done.

File: lottie/lib/lottie/LottieAnimation.qml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.9
+import vpn.mozilla.lottie 1.0
+
+Item {
+    id: lottieItem
+
+    // The source can be a QRC URL only.
+    property alias source: lottiePrivate.source
```
[the remark this answers is not in the document]
Comment (bakulf) [reply lottie/lib/lottie/LottieAnimation.qml:12]: done.

**Review by mkestler-mozilla — APPROVED** [review]

**Pull request merged**